academic-services/knowledge-exploration-service/reference-makes-command-line-tool.md

```diff
@@ -171,37 +197,288 @@ The input entities file Url prefix.
+
+## Evaluate command
+
+Evaluates a KES query expression and returns the top matching entities in the index(es).
+
+```cmd
+kesm Evaluate --KesQueryExpression
+              --IndexFilePaths
+              [--Attributes]
+              [--Skip]
+              [--Take]
+              [--OrderBy]
+              [--OrderByDescending]
+              [--Timeout]
+```
+
+### Evaluate required parameters
+
+`--KesQueryExpression`
+
+The KES [query expression](.\concepts-query-expressions.md) to use for selecting entities from the index(es).
+
+`--IndexFilePaths`
+
+The file path expression for specifying which index file(s) to use. Use wild card to specify multiple indexes e.g. './index.*.kes'
+
+### Evaluate optional parameters
+
+`--Attributes`
+
+A list of entity attributes to be included in the result set, seperated by ','. Use '*' for all attributes. Defaults to 0.
+
+`--Offset`
+
+The number of top entities to be skipped/excluded in the result set. Defaults to 0.
+
+`--Count`
+
+The number of top interpretations to be included in the result set. Defaults to 5.
+
+`--OrderBy`
+
+Name of the entity attribute to use for sorting/ordering the entities in the result set. By default, entities are sorted by descending entity weight (static rank). Defaults to "weight"
+
+`--OrderByDescending`
+
+The direction for sorting entities. By default, entities are sorted by descending entity weight (static rank). Defaults to true.
+
+`--Timeout`
+
+Maximum amount of time in milliseconds allowed for the command to complete before aborting the command. Use 0 to disable timeout. Defaults to 0.
+
+## Histogram command
+
+Calculates distinct/total entity attribute counts and top attribute values for entities matching a query expression.
+
+```cmd
+kesm Histogram --KesQueryExpression
+               --IndexFilePaths
+               [--Attributes]
+               [--Skip]
+               [--Take]
+               [--OrderBy]
+               [--OrderByDescending]
+               [--Timeout]
+```
+
+### Histogram required parameters
+
+`--KesQueryExpression`
+
+The KES [query expression](.\concepts-query-expressions.md) to use for selecting entities from the index(es).
+
+`--IndexFilePaths`
+
+The file path expression for specifying which index file(s) to use. Use wild card to specify multiple indexes e.g. './index.*.kes' Histogram requires at least one index.
+
+### Histogram optional parameters
+
+`--Attributes`
+
+A list of entity attributes, seperated by ','. Use '*' for all attributes. Histogram will generate total count, distinct count, and top values for the select attributes for entities specified in the KesQueryExpression. Defaults to "*".
+
+`--Offset`
+
+The number of top entity attribute values to be skipped/excluded in the result set. Defaults to 0.
+
+`--Count`
+
+The number of top entity attribute values to be included in the result set. Defaults to 5.
```

Comment:
>5 [](start = 88, length = 1)

10?

Reply:
It's default to 5 for kesm. 

---
In reply to: [465230966](https://github.com/MicrosoftDocs/microsoft-academic-services/pull/172#discussion_r465230966) [](ancestors = 465230966)